File: 27.移除元素.py
#
# @lc app=leetcode.cn id=27 lang=python3
#
# [27] 移除元素
#

# @lc code=start
# class Solution:
#     def removeElement(self, nums: List[int], val: int) -> int:
#         slow=0
#         # fast找新数组元素，slow是新数组下标
#         for fast in range(len(nums)):
#             if nums[fast]!=val:
#                 nums[slow]=nums[fast]
#                 slow+=1
#             fast+=1
#         return slow
class Solution:
    def removeElement(self,nums:List[int],val:int)->int:
        slow=0
        for fast in range(len(nums)):
            if nums[fast]!=val:
                nums[slow]=nums[fast]
                slow+=1
            fast+=1
        return slow



# @lc code=end

File: 27.移除元素.cpp
/*
 * @lc app=leetcode.cn id=27 lang=cpp
 *
 * [27] 移除元素
 */

// @lc code=start
// class Solution {
// public:
//     int removeElement(vector<int>& nums, int val) {
//         int slow=0;
//         // int n=nums.size(); 注释掉比不注释慢很多！！
//         for (int i =0; i<nums.size();++i){
//             if (nums[i]!=val){
//                 nums[slow]=nums[i];
//                 slow++;
//             }            
//         }
//         return slow;
//     }
// };

class Solution{
public:
    int removeElement(vector<int>& nums, int val){
        int slow=0;
        for (int fast=0;fast<nums.size();fast++){
            if (nums[fast]!=val){
                nums[slow]=nums[fast];
                slow++;
            }
        }
        return slow;
    }
};
// @lc code=end

